lib/pact/provider_verifier/add_interaction_attributes_header.rb

```diff
@@ -0,0 +1,39 @@
+require 'delegate'
+
+module Pact
+  module ProviderVerifier
+    class RequestDelegate < SimpleDelegator
+      def initialize request, extra_rack_headers
+        super(request)
+        @extra_rack_headers = extra_rack_headers
+      end
+
+      def headers
+        __getobj__().headers.merge(@extra_rack_headers)
+      end
+
+      def method
+        __getobj__().method
+      end
+    end
+
+    class AddInteractionAttributesHeader
+      def self.call(request, interaction)
+        extra_rack_headers = {
+          "X_PACT_DESCRIPTION" => interaction.description
+        }
+
+        if interaction.provider_states
```

Comment:
provider_states is always set, even if parsing a v2 spec. which means this block of code always runs, and the block beneath doesn't

https://github.com/pact-foundation/pact-provider-verifier/blob/ea3f24710e1ab702205099619cea8fa9c79e60b0/lib/pact/provider_verifier/set_up_provider_state.rb#L67

with a v2 spec, we get a change in behaviour


```
{"rack.version"=>[1, 3], "rack.input"=>#<StringIO:0x0000000139657a90>, "rack.errors"=>#<StringIO:0x0000000139658710>, "rack.multithread"=>true, "rack.multiprocess"=>true, "rack.run_once"=>false, "REQUEST_METHOD"=>"GET", "SERVER_NAME"=>"example.org", "SERVER_PORT"=>"80", "QUERY_STRING"=>"", "PATH_INFO"=>"/somestate", "rack.url_scheme"=>"http", "HTTPS"=>"off", "SCRIPT_NAME"=>"", "CONTENT_LENGTH"=>"0", "rack.test"=>true, "REMOTE_ADDR"=>"127.0.0.1", "SERVER_PROTOCOL"=>"HTTP/1.0", "HTTP_VERSION"=>"HTTP/1.0", "X_PACT_DESCRIPTION"=>"Provider state success", "X_PACT_PROVIDER_STATES"=>[{"name"=>"There is a greeting", "params"=>{}}], "HTTP_HOST"=>"example.org", "HTTP_COOKIE"=>""}
```

`"X_PACT_DESCRIPTION"=>"Provider state success"`
`"X_PACT_PROVIDER_STATES"=>[{"name"=>"There is a greeting", "params"=>{}}]`

before change

```
{"rack.version"=>[1, 3], "rack.input"=>#<StringIO:0x0000000122dd5248>, "rack.errors"=>#<StringIO:0x0000000122dd5838>, "rack.multithread"=>true, "rack.multiprocess"=>true, "rack.run_once"=>false, "REQUEST_METHOD"=>"GET", "SERVER_NAME"=>"example.org", "SERVER_PORT"=>"80", "QUERY_STRING"=>"", "PATH_INFO"=>"/somestate", "rack.url_scheme"=>"http", "HTTPS"=>"off", "SCRIPT_NAME"=>"", "CONTENT_LENGTH"=>"0", "rack.test"=>true, "REMOTE_ADDR"=>"127.0.0.1", "SERVER_PROTOCOL"=>"HTTP/1.0", "HTTP_VERSION"=>"HTTP/1.0", "X_PACT_PROVIDER_STATES"=>[{"name"=>"There is a greeting"}], "HTTP_HOST"=>"example.org", "HTTP_COOKIE"=>""}
```

`"X_PACT_PROVIDER_STATES"=>[{"name"=>"There is a greeting"}]`